modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -451,6 +470,138 @@ TensorMechanicsAction::actOutputGeneration()
+      std::vector<std::string> add_list;
+
+      if (mat_params.isParamValid("ad_props_out") && mat_params.isParamValid("reg_props_in") &&
+          _use_ad)
+      {
+        remove_list = mat_params.get<std::vector<std::string>>("reg_props_in");
+        add_list = mat_params.get<std::vector<std::string>>("ad_props_out");
+      }
+      if (mat_params.isParamValid("ad_props_in") && mat_params.isParamValid("reg_props_out") &&
+          !_use_ad)
+      {
+        remove_list = mat_params.get<std::vector<std::string>>("ad_props_in");
+        add_list = mat_params.get<std::vector<std::string>>("reg_props_out");
+      }
+
+      // These vectors are the same size as checked in MaterialConverter
+      for (unsigned int index = 0; index < remove_list.size(); index++)
+        remove_add_map.emplace(remove_list[index], add_list[index]);
+    }
+  }
+  // All the materials have been accounted for, now remove or add parts
+
+  // Remove names which aren't eigenstrains (converter properties)
+  for (auto remove_add_index : remove_add_map)
+  {
+    const bool is_in = eigenstrain_set.find(remove_add_index.first) != eigenstrain_set.end();
+    if (is_in)
+    {
+      eigenstrain_set.erase(remove_add_index.first);
+      eigenstrain_set.insert(remove_add_index.second);
+    }
+  }
+  for (auto index : remove_reduced_set)
+  {
+    if (eigenstrain_set.find(index) != eigenstrain_set.end())
+      eigenstrain_set.erase(index);
+  }
+
+  // Compare the blockIDs set of eigenstrain names with the vector of _eigenstrain_names for the
+  // current subdomainID
+  std::set_union(eigenstrain_set.begin(),
+                 eigenstrain_set.end(),
+                 _eigenstrain_names.begin(),
+                 _eigenstrain_names.end(),
+                 std::inserter(verified_eigenstrain_names, verified_eigenstrain_names.begin()));
+
+  // Ensure the eigenstrain names previously passed include any missing names
+  _eigenstrain_names.resize(verified_eigenstrain_names.size());
+  std::copy(verified_eigenstrain_names.begin(),
+            verified_eigenstrain_names.end(),
+            _eigenstrain_names.begin());
+
+  // Since the automatic eigenstrain names have been collected, return the list
+  auto list_subdomains = _problem->mesh().meshSubdomains();
+  // If no defined blocks
+  if (list_subdomains.size() == 1 && *list_subdomains.begin() == 0)
+    Moose::out << COLOR_CYAN << "*** Automatic Eigenstrain Names ***\n"
+               << _name << ": " << Moose::stringify(_eigenstrain_names) << COLOR_DEFAULT << "\n";
+  // Only print once for multiple blocks
+  else
+    for (auto currentNum : _subdomain_ids)
+    {
+      if (currentNum == *list_subdomains.begin())
+        Moose::out << COLOR_CYAN << "*** Automatic Eigenstrain Names ***\n";
+      Moose::out << _name << ": " << Moose::stringify(_eigenstrain_names) << "\n";
+      if (currentNum == *list_subdomains.rbegin())
+        Moose::out << COLOR_DEFAULT << "\n";
+    }
```

Comment:
why is this subdomains business still here? I do not get it. Wherever you are in that loop you _always_ just print `_eigenstrain_names`. What's the point of looping?! I commented on this in a previous review with a code suggestion that removed all this. I did not see a reply why that should _not_ be done.

Reply:
This was to only print out the header once, followed by the blocks and associated eigenstrains.  I've used a regex within the tests